code/modules/mob/dead/observer/observer.dm

```diff
@@ -865,3 +866,47 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		spawners_menu = new(src)
 
 	spawners_menu.ui_interact(src)
+
+
+/mob/dead/observer/proc/become_mouse()
+	set name = "Become mouse"
+	set category = "Ghost"
+	var mouse_respawn_time = 5
+
+	if(GLOB.disable_player_mice)
+		src << "<span class='warning'>Spawning as a mouse is currently disabled.</span>"
+		return
+
+	var/area/T = get_area(src)
+	if(T.noteleport)
+		src << "<span class='warning'>You may not spawn as a mouse on this Z-level.</span>"
+		return
+
+	var/timedifference = world.time - client.time_died_as_mouse
+	if(client.time_died_as_mouse && timedifference <= mouse_respawn_time * 600)
```

Comment:
```suggestion
	if(client.time_died_as_mouse && timedifference <= 5 MINUTES)
```

Reply:
Are you sure it wouldnt be more flexible to have a var so admins can adjust the respawn timer if they want on the fly? I could add it to the globals.

Reply:
Then make it a global var.
place `GLOBAL_VAR_INIT(mouse_respawn_time, 5 MINUTES)` in either `code/_globalvars/misc.dm` or a line above `become_mouse()` function.
And make sure to add in the `GLOB.` prefix path whenever you reference the global var (eg : `GLOB.mouse_respawn_time`)